app/test/search/backend_test.dart

```diff
@@ -69,5 +69,12 @@ void main() {
         'topic:widget abc',
       );
     });
+
+    test('query with topic shortcut `#`', () {
+      expect(
+        canonicalizeSearchForm(_parse('#widget #testing'))?.query,
+        'topic:widget topic:testing',
+      );
```

Comment:
Can we have a test case where I search for `"#hash"` -- and would that work if I actually want to search for a keyword that starts with `#` ?

Reply:
`#` is by default removed from the content at indexing and from queries (used as a separator).

The end result of this change is that `#hash` search from https://pub.dev/packages?q=%23hash gets redirected to https://pub.dev/packages?q=topic%3Ahash and yeah, we can add a test for that.